File: tests/fixtures/ruby/sample.rb
require_relative "shared/helper"

module Acme
  module Shared
    module Formatter
      def format(name)
        name.strip
      end
    end
  end
end

class BaseService
end

class UserService < BaseService
  include Acme::Shared::Formatter

  def greet(name)
    helper = Helper.new
    helper.decorate(format(name))
  end
end

File: tests/fixtures/php/sample.php
<?php

namespace Acme\Sample;

use Acme\Shared\Helper as SharedHelper;
use Acme\Shared\Logger;
use function Acme\Shared\format_name;

interface Greeter
{
    public function greet(string $name): string;
}

class BaseService
{
}

class UserService extends BaseService implements Greeter
{
    private string $name = '';

    public function greet(string $name): string
    {
        $helper = new SharedHelper();
        Logger::write(format_name($name));
        return $helper->decorate($name);
    }
}
